src/util.rs

```diff
@@ -33,6 +33,10 @@ pub trait StrIteratorExt {
     fn chain_one<S>(self, s: S) -> Vec<String>
     where
         S: fmt::Display;
+
+    fn chain_many<S>(self, s: Vec<S>) -> Vec<String>
```

Comment:
I think this loses the ergonomics! The magic of `chain_one` is that you can pass any stringable type and it will get stringified, without you needing to specifically type `to_string`. The `chain_mainy` method does not have the magic, as it constrains all types in the vector to be the same `S`. And in your usage this requires calling `to_string()` on each element of the vector—at which point you can just use the `chain` function that’s built in to the standard `Iterator` trait.

It’s been a minute since I’ve looked at this crate, but IIRC the idiom is to chain `chain_one` calls, as in `path.chain_one(“a”).chain_one(some_string)`. Or you can stick with the batch approach you have, calling `to_string` on each element in the collection, but use the built-in `chain` method. You should also be able to use an array rather than a vec!

Reply:
Ah, I see, `chain_one` and `chain_many` return `Vec`s rather than iterators, and so you can’t actually chain the calls themselves.

Reply:
Actually, it does appear to support chained calls `chain_one(...).chain_one(...)`. I've pushed an update that removes `chain_many` in favor of this.